### DISCUSSION TITLE
CSR In Analog or Hybrid Approaches

### DISCUSSION THREAD

**Discussion by wilsonsergio2500** [analogjs/analog#866] [Q&A] [answered]
Thank you for the great work done in Analog and building this great community :heart_eyes:

Just very recently, I have been exploring the Analog architecture and have bootstrapping a few examples. - so far so love! However, the concept that it hasn't been evident to me, at least in this early exploration, is how Analog could induce Client Side Rendering within its architecture.

One one hand, we have NextJS where the use of 'use client' aids us in rendering such a component as client component. On the other hand we have Astro, which uses the concept of  island, and marking them as client help us define one as a client component. I do see that Analog supports an Astro integration which is very compelling.

However, within Analog, how does a hybrid - SSR + CSR - approach combination comes about. Say for instance, if we rather induce client interactivity through a Vanilla Angular stand alone component (That may render in one of the server routes). I apologize if this has not been evident and It should. It is likely nonetheless, that for others this does not become evident at first glance.

I appreciate any further discussion when it comes to Analog approaches for CSR.
Thank you once again

**Comment by brandonroberts** [comment]
Thanks @wilsonsergio2500 

If you want to CSR-only some pages, you can do that through the `main.server.ts`. We have an open issue on a better API for this in the future 

https://github.com/analogjs/analog/issues/810

**Reply by wilsonsergio2500** [reply, marked as answer]
thank you Brandon I will go ahead and play with this  😃
